Add PowerShell example

Adam-S-Daniel opened the pull request:
# Title

Add PowerShell example

## Summary

Created a section showing how to "_check for a minimum-required .NET Framework version by querying the registry in PowerShell_" with a single-line example command.

Adam-S-Daniel committed:
Add PowerShell example

dnfclas commented:
@__geeeyetee__,
Thanks for your contribution.
To ensure that the project team has proper rights to use your work, please complete the Contribution License Agreement at https://cla2.dotnetfoundation.org.

It will cover your contributions to all .NET Foundation-managed open source projects.
_Thanks,
.NET Foundation Pull Request Bot_

dnfclas commented:
__@geeeyetee__, thanks for signing the contribution license agreement. We will now validate the agreement and then the pull request.
<br />_Thanks, .NET Foundation Pull Request Bot_

merriemcgaw reviewed (commented):
You need to break out the Win10 Release key values for each version here just like #2 above does. We also need to check for greater than or equal to, the snippet checks greater than.

BillWagner reviewed (dismissed):
@geeeyetee If you can update this based on @merriemcgaw 's comments, I'll :shipit: . If you don't have time, I'll incorporate the requested changes and merge this. Let me know your preference.

Thanks for the contribution, we really appreciate it.

geeeyetee commented:
@BillWagner I won't have time; please go ahead and incorporate the requested changes. Thanks!

> On Jul 11, 2017, at 11:13 AM, Bill Wagner <notifications@github.com> wrote:
> 
> @BillWagner requested changes on this pull request.
> 
> @geeeyetee If you can update this based on @merriemcgaw 's comments, I'll  . If you don't have time, I'll incorporate the requested changes and merge this. Let me know your preference.
> 
> Thanks for the contribution, we really appreciate it.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

BillWagner committed:
Update based on feedback

Adding Windows 10 specific values, and changing -ge to -gt

BillWagner commented:
@mairaw Since I made the commit based on @merriemcgaw 's feedback, can you review?

mairaw commented on docs/framework/migration-guide/how-to-determine-which-versions-are-installed.md:
```diff
@@ -150,6 +151,29 @@ Users can install and run multiple versions of the .NET Framework on their compu
+
+    |Version|Minimum value of the Release DWORD|
+    |-------------|--------------------------------|
+    |.NET Framework 4.5|378389|
```
if @merriemcgaw was referring to https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed#to-find-net-framework-versions-by-viewing-the-registry-net-framework-45-and-later, when she said #2, I think we're missing a few values, no? Like 378758.

geeeyetee replied:
Because this example checks "for a minimum-required .NET Framework version", its reference table deliberately consolidates by Framework version, omitting distinct version numbers for different operating systems.

mairaw replied:
so should we consolidate for the other versions of the Framework? @BillWagner @geeeyetee 
We're listing the different OS versions for 4.7, 4.6.1, 4.6.2, etc.

geeeyetee replied:
Addressed in https://github.com/dotnet/docs/commit/7394864f310d22ca00e75053bbe8eae37b901f65

mairaw commented on docs/framework/migration-guide/how-to-determine-which-versions-are-installed.md:
````diff
@@ -150,6 +151,29 @@ Users can install and run multiple versions of the .NET Framework on their compu
+    |.NET Framework 4.7 installed on Windows 10 Creators Update|460798|
+    |.NET Framework 4.7 installed on all other Windows OS versions|460805|
+
+```PowerShell
````
Wondering if we move the example up right after the sentence we introduce the example and then show the table with the different values you can use after the example.

geeeyetee replied:
That would be fine with me; good idea.

BillWagner committed:
respond to feedback

@mairaw this should be ready to go.

Adam-S-Daniel reviewed (commented):
@mairaw @BillWagner = Agreed. Looks good to me.

mairaw commented on docs/framework/migration-guide/how-to-determine-which-versions-are-installed.md:
````diff
@@ -156,6 +156,10 @@ Users can install and run multiple versions of the .NET Framework on their compu
+```PowerShell
+Get-ChildItem "hklm:SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\" | Get-ItemPropertyValue -Name Release | % { $_ -ge 394802 } 
+```
+
````
weird, the table now is broken and tokens didn't resolve: 

![image](https://user-images.githubusercontent.com/12971179/28850331-d20baf60-76d0-11e7-9fa1-2e0f26607736.png)

geeeyetee replied:
Addressed in commit https://github.com/dotnet/docs/commit/7394864f310d22ca00e75053bbe8eae37b901f65. The fix was to indent the PowerShell snippet.

mairaw commented on docs/framework/migration-guide/how-to-determine-which-versions-are-installed.md:
````diff
@@ -156,6 +156,10 @@ Users can install and run multiple versions of the .NET Framework on their compu
 
 - The following example checks the value of the `Release` keyword to determine whether .NET Framework 4.6.2 or higher is installed (returning `True` if it is and `False` otherwise). You can substitute another value listed in the table to check for a different minimum-required .NET Framework version.
 
+```PowerShell
````
I'd move the last sentence from the previous paragraph to after the code to introduce the table

geeeyetee replied:
Addressed in commit https://github.com/dotnet/docs/commit/7394864f310d22ca00e75053bbe8eae37b901f65

BillWagner committed:
fixing table formatting.

BillWagner commented:
@mairaw I'm not sure I understand you're last comment. What should change there?

Adam-S-Daniel committed:
Improvements to Framework version PowerShell query

- Consolidated framework version table, removing OS version subdivisions.
- Added explanatory phrase "regardless of Windows OS version". 
- Moved sentence about substitution to below the framework version table.
- Fixed broken table rendering.

geeeyetee commented:
@BillWagner, I hope I didn't step on anything from https://github.com/dotnet/docs/pull/2545/commits/f33d91ca2ca0f8e406643c944b15670f815ce927, which you committed a couple of minutes before my commit https://github.com/dotnet/docs/pull/2545/commits/7394864f310d22ca00e75053bbe8eae37b901f65. I believe mine addresses all of the issues @mairaw raised, as I understood them.

geeeyetee commented:
@mairaw  - Any remaining concerns?

mairaw committed:
minor changes

mairaw committed:
Merge branch 'master' into patch-1

mairaw commented:
Apologies on the delay for getting back to you @geeeyetee. It looked great. I did some minor edits and changed the metadata to reflect this important change. I'll merge as soon as CI passes. Thank you!

mairaw committed:
fixed metadata
